src/freadR.c

```diff
@@ -161,11 +163,34 @@ SEXP freadR(
     readInt64As = CT_FLOAT64;
   } else STOP("Invalid value integer64='%s'. Must be 'integer64', 'character', 'double' or 'numeric'", tt);
 
-  colClassesSxp = colClassesArg;   // checked inside userOverride where it is used.
+  colClassesSxp = colClassesArg;
 
-  if (!isNull(selectArg) && !isNull(dropArg)) STOP("Use either select= or drop= but not both.");
   selectSxp = selectArg;
   dropSxp = dropArg;
+  selectColClasses = false;
+  if (!isNull(selectSxp)) {
+    if (!isNull(dropSxp)) STOP("Use either select= or drop= but not both.");
+    if (isNewList(selectArg)) {
+      if (!isNull(colClassesSxp))
+        STOP("select= is type list for specifying types in select=, but colClasses= has been provided as well. Please remove colClasses=.");
+      if (!length(getAttrib(selectArg, R_NamesSymbol)))
+        STOP("select= is type list so expecting list(type1=cols1, type2=cols2, ...) but has no names");
```

Comment:
"but the input has no names"

Reply:
Good spot. I'll reword that.  (The "no names" there refers to the names of the select= argument (the type names) rather than the input data.)